From 4de19d56545a3f592eb64ab7afc20e1c4e93ad53 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Wed, 28 Mar 2018 14:24:43 +0200
Subject: [PATCH 04/23] discover/grub: Allow to choose a different BLS
 directory

The default path to search for BootLoaderSpec configuration files is
/loader/entries but in some setups a different directory may be used.

So allow this to be chosen by using a blsdir GRUB environment variable.

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
Signed-off-by: Samuel Mendoza-Jonas <sam@mendozajonas.com>
(cherry picked from commit d01dfd5a6ca8283939265875dc69d665f955aba2)
Signed-off-by: Klaus Heinrich Kiwi <klaus@linux.vnet.ibm.com>
---
 discover/grub2/blscfg.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/discover/grub2/blscfg.c b/discover/grub2/blscfg.c
index 0f69f7e..02ac621 100644
--- a/discover/grub2/blscfg.c
+++ b/discover/grub2/blscfg.c
@@ -180,6 +180,7 @@ int builtin_blscfg(struct grub2_script *script,
 	struct conf_context *conf;
 	struct bls_state *state;
 	char *buf, *filename;
+	const char *blsdir;
 	int n, len, rc = -1;
 
 	conf = talloc_zero(dc, struct conf_context);
@@ -191,12 +192,16 @@ int builtin_blscfg(struct grub2_script *script,
 	conf->process_pair = bls_process_pair;
 	conf->finish = bls_finish;
 
-	n = parser_scandir(dc, BLS_DIR, &bls_entries, bls_filter, bls_sort);
+	blsdir = script_env_get(script, "blsdir");
+	if (!blsdir)
+		blsdir = BLS_DIR;
+
+	n = parser_scandir(dc, blsdir, &bls_entries, bls_filter, bls_sort);
 	if (n <= 0)
 		goto err;
 
 	while (n--) {
-		filename = talloc_asprintf(dc, BLS_DIR"/%s",
+		filename = talloc_asprintf(dc, "%s/%s", blsdir,
 					   bls_entries[n]->d_name);
 		if (!filename)
 			break;
-- 
2.17.1

